spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/EntityPropertyValueProvider.java

```diff
@@ -52,13 +62,22 @@
 	private <T> T readSingleWithConversion(DatastorePersistentProperty persistentProperty) {
 		Object val = this.entity.getValue(persistentProperty.getFieldName()).get();
 		Class<?> targetType = persistentProperty.getType();
-		if (val == null) {
-			return null;
-		}
-		if (ClassUtils.isAssignable(targetType, val.getClass())) {
+
+		if (val == null || ClassUtils.isAssignable(targetType, val.getClass())) {
 			return (T) val;
```

Comment:
This will not work for a case of `Object` -> `String`, where `String` is a JSON string that you need to convert.

Move it to the final case.